### PR TITLE
Fix (Carousel block): links and buttons behavior on iOS

### LINKED ISSUE #3011: iOS x Carousel issue - When there’s an existing Carousel block in the page and you click any button or link, it doesn’t respond on the first click
**Describe the bug**
- When clicking a button block with link when there's a carousel block in the page, it only responds to the second click.
- Findings: Make sure to hard refresh browser when testing
- Issue on frontend
- Issue on iOS only
- Happens to browsers such as: Safari, Chrome, etc.
- Here's a [test page](https://gambitsandbox.instawp.xyz/andrea/wp-admin/post.php?post=188&action=edit) on Sandbox

**To Reproduce**
1. Add a new page OR skip step 2-6 and open the test page mentioned above
2. Add a Carousel block
3. Add an Image block on each slider
4. Add link on each nested Image block (optional)
5. Add a Button block
6. Assign a link for the Button block
7. Save page
8. View frontend
9. Try clicking the Button block
10. See that it doesn't respond (issue)
11. Repeat step 9
12. See that it now responds upon second click

**Expected behavior**
- When there's an existing Carousel block and when you click any button or link, it should respond upon first click already.

**Screenshots**

https://github.com/gambitph/Stackable/assets/103395655/a622ec7c-15e6-4236-ada4-3e563e32afb1

**Smartphone:**
 - Device: iPhone 12 Pro (no specific iOS device)
 - OS: iOS 17.2 (no specific OS)
 - Browser: e.g. Safari, Chrome

### PR DESCRIPTION
**Pull request by kaeizen**
fixes #3011 

The issue is caused by the touch events and is a common issue in iOS. Similar issues found:
https://stackoverflow.com/questions/7573126/ipad-iphone-double-click-problem
https://github.com/wp-media/wp-rocket/issues/3142

### CODE REVIEW

File: src/block/carousel/frontend-carousel.js [hunk 1]
```diff
@@ -424,6 +444,32 @@ class _StackableCarousel {
...
+
+		if ( target.closest( '.stk-block-carousel__slider' ) ) {
+			e.preventDefault()
+			this.hasTouched = false
+			return
+		}
+
+		// Manually trigger click event on links and buttons
+		if ( target.tagName.toLowerCase() === 'a' ||
+			target.tagName.toLowerCase() === 'button' ||
```
Comment (bfintal) [inline src/block/carousel/frontend-carousel.js]:
```suggestion
		if ( target.tagName === 'A' ||
			target.tagName === 'BUTTON' ||
```

**Review by bfintal — COMMENTED** [review]
@mxkae would it work if we simply changed the `touch*` events to `pointer*` events?

**Comment by kaeizen** [comment]
@bfintal I tried using pointer events instead of touch events. Buttons and links still won't respond on first click.

https://github.com/gambitph/Stackable/assets/98727316/640ca351-5c2d-44d1-ae56-d8d25f92ea75

**Comment by andeng1106** [comment]
@mxkae Issue is now okay. But there's a remaining issue:

- Links set on inner column and image inside the Carousel block is not working when clicking

https://github.com/gambitph/Stackable/assets/103395655/ee3dded8-fcef-4f56-bc04-3d3f2539bede

**Pull request merged**